generators/E906LegacyGen/SQPrimaryVertexGen.h

```diff
@@ -16,23 +16,21 @@ from Kun to E1039 experiment in Fun4All framework
 #include <TF2.h>
 #include <TVector3.h>
 #include <TH1F.h>
+#include <fun4all/SubsysReco.h>
 #include "SQBeamlineObject.h"
 
 class PHCompositeNode;
 class SQBeamlineObject;
 
-class SQPrimaryVertexGen
+class SQPrimaryVertexGen: public SubsysReco
```

Comment:
This class became a SubsysReco child class here, but it is not complete since its process_event() function hasn't been defined.  How do you intend to use this class?

Reply:
I intend to make this SubsysReco child class because I wanted to set the beam profile in the Fun4Sim macro. So it needs to be registered into the Fun4All server.

Reply:
The Fun4Sim line is 
https://github.com/E1039-Collaboration/e1039-analysis/blob/c697b092036563e1dd0afb6cb09385da70ff26f8/SimChainDev/Fun4Sim.C#L76-L79

Reply:
It can be registered but does nothing per event because its process_event() is not implemented.  Saying in other words, what calls "SQPrimaryVertexGen::traverse()" when?

Reply:
Yes, it is not doing anything per event. I just use it to set the beam-profile at the beginning of the run via IntRun() function of Subsysreco.

The SQPrimaryVertexGen::traverse() is called in the process_event of SQPrimaryParticleGen for setting up the vertices.

Reply:
Since it is not the standard usage of SubsysReco module, the current code is (and will be) confusing.  We can rather make SQPrimaryVertexGen a member of SQPrimaryParticleGen, and initialize it in SQPrimaryParticleGen::InitRun().

Reply:
The only reason for making it Subysreco class is to initialize the beam profile for now. I am thinking of this vertex generator code to make stand alone so that it can be used in the other minimum bias generator for eg. particle guns, simple event generators in the future.

Reply:
I understand the purpose (making it standalone) and agree it is better.  But I mean the method is not suitable at present in several lines of this class.

If this class is SubsysReco, it should have its own "process_event()" (which generate a vertex position in data node).  It should not be called by another SubsysReco like SQPrimaryParticleGen.

Reply:
Yes, I think I agree with you for most of the part. It would be better if I can make the vertexgen code more efficient and move it to the another location in such a way that the particle guns and other event generators can also use it. I will keep on working on it but for Monday, I just want to keep as it is (if you are ok with it).